drizzlepac/alignimages.py

```diff
@@ -527,28 +524,33 @@ def run_align(input_list, archive=False, clobber=False, debug=False, update_hdr_
                         if fit_quality < 5:
                             if fit_quality == 1:  # valid, non-comprimised solution with total rms < 10 mas...go with this solution.
                                 best_fit_rms = fit_rms
-                                for item in imglist:
-                                    item.best_meta = item.meta.copy()
+
+                                best_imglist = copy.deepcopy(imglist)
+
                                 best_fit_status_dict = fit_status_dict.copy()
+                                best_fit_qual = fit_quality
                                 break  # break out of while loop
                             elif fit_quality < best_fit_qual:  # better solution found. keep looping but with the better solution as "best" for now.
                                 log.info("Better solution found!")
                                 best_fit_rms = fit_rms
-                                for item in imglist:
-                                    item.best_meta = item.meta.copy()
+
+                                best_imglist = copy.deepcopy(imglist)
+
                                 best_fit_status_dict = fit_status_dict.copy()
                                 best_fit_qual = fit_quality
                             elif fit_quality == best_fit_qual:  # new solution same level of fit_quality. Choose whichever one has the lowest total rms as "best" and keep looping.
                                 if best_fit_rms >= 0.:
                                     if fit_rms < best_fit_rms:
                                         best_fit_rms = fit_rms
-                                        for item in imglist:
-                                            item.best_meta = item.meta.copy()
+
+                                        best_imglist = copy.deepcopy(imglist)
+
                                         best_fit_status_dict = fit_status_dict.copy()
+                                        best_fit_qual = fit_quality
                             else:  # new solution has worse fit_quality. discard and continue looping.
                                 continue
                             # preserve best fit solution so that it can be inserted into a reinitialized imglist next time through.
-                            temp_imglist = copy.deepcopy(imglist)
+                            best_imglist = copy.deepcopy(imglist)
```

Comment:
I am not sure why line #553 is needed here.  The only case not covered by the if/elif/else is when fit_quality >= 5.  When fit_quality >= 5, the best_imglist is an empty list, and if this variable is then used later and is not checked for being empty (before it is filled at all), then this could cause a problem.

Reply:
It was originally intended to initialize the 'best' solution as needed for the now replaced/deleted `temp_imglist` list.  Additional testing demonstrated that it is not actually needed and has been removed in commit 842988d.